stl/inc/filesystem

```diff
@@ -1650,6 +1650,15 @@ namespace filesystem {
         return _Val;
     }
 
+    _NODISCARD inline bool _Relative_path_contains_root_name(const path& _Path) {
+        for (const auto& _File_name : _Path.relative_path()) {
+            if (!_Parse_root_name(_File_name.native()).empty()) {
+                return true;
+            }
+        }
+        return false;

```

Comment:
This should be doable without N extra memory allocations. How about something like:

```
// insert this function above:
bool _Starts_with_root_name(const wchar_t* const _First, const wchar_t* const _Last) {
    return _Find_root_name(_First, _Last) != _First;
}

// here:
auto& _Native = _Path.native();
auto _First         = _Native.data();
const auto _Last          = _First + _Native.size();
_First = _Find_relative_path_end(_First, _Last);
while (_First != _Last) {
    const auto _Next = _STD find_if(_First, _Last, _Is_slash);
    if (_Starts_with_root_name(_First, _Next)) {
        return true;
    }

    _First = _STD find_if_not(_Next, _Last, _Is_slash);
}

return false;
```

(Note that I haven't proven the above correct so please double check :) )

Reply:
@BillyONeal, would you be happy with filing a `performance` issue to investigate avoiding memory allocations here? I would prefer to separate this LWG issue resolution (where the loop's correctness is easy to verify when expressed like this) from optimization work involving significant runtime logic.

The core algorithm is already iterating through `path` so it's not like this PR is introducing allocations to a codepath that was previously allocation-less.

Reply:
Hmm that's true but it is still introducing N more allocations. (I did not mark 'request changes' over it though, just a suggestion)

Reply:
(To be clear, I'm proposing a change to `_Relative_path_contains_root_name` which is new in this PR, not suggesting any changes to the existing path iterator loop below)

Reply:
Yep, I know. The change is introducing more allocations but also more conformance; I think it's reasonable to merge now. I'm wary of delaying a targeted, clearly correct PR with optimization work that will require more testing and thinking, and I'm not too worried about the performance implications of shipping the naive loop. (In more concrete terms, if we can merge this PR *this week*, it can get into VS 2019 16.8 Preview 1; any later and it will be Preview 2.)

Reply:
I filed #937 to track this improvement.